python/tvm/driver/tvmc/compiler.py

```diff
@@ -0,0 +1,407 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Provides support to compile networks both AOT and JIT.
+"""
+import argparse
+import logging
+import tarfile
+from pathlib import Path
+
+import tvm
+from tvm import autotvm
+from tvm import relay
+from tvm._ffi.runtime_ctypes import TVMContext
+from tvm.contrib import cc
+from tvm.contrib import util
+from tvm.relay.op.contrib import get_pattern_table
+
+from . import common, frontends
+from .main import register_parser
+
+# A dictionary of target aliases to simplify the command lines provided by end users
+TARGET_ALIASES = {
+    "aarch64": "llvm -device=arm_cpu -mtriple=aarch64-linux-gnu -mattr=+neon"
+}
+
+#  A list of valid targets (including aliases) to be used in "--target"
+VALID_TARGETS = ["aarch64", "llvm"]
+
+DEFAULT_TARGET = "llvm"
+DUMP_FORMATS = ["relay", "ll", "asm"]
+
+
+def parse_target(targets_str):
+    """ Parsing function for comma separated target syntax. """
+    targets = targets_str.split(",")
+    for target in targets:
+        if target not in VALID_TARGETS:
+            raise argparse.ArgumentTypeError(f"unrecognized target: {target}")
+    return targets
+
+
+@register_parser
+def add_compile_parser(subparsers):
+    """ Include parser for 'compile' subcommand """
+
+    parser = subparsers.add_parser("compile", help="compile a model")
+    parser.set_defaults(func=drive_compile)
+    parser.add_argument(
+        "--cross-compiler",
+        default="",
+        help="the cross compiler to use to generate target libraries",
+    )
+    parser.add_argument(
+        "--dump-codegen", default="", choices=DUMP_FORMATS, help="dump generated code"
+    )
+    parser.add_argument(
+        "--language",
+        choices=frontends.get_frontends(),
+        help="specify input language",
+    )
+    parser.add_argument(
+        "--input-shape",
+        type=common.parse_input_shapes,
+        metavar="INPUT_SHAPE,[INPUT_SHAPE]...",
+        help="for pytorch, e.g. '(1,3,224,224)'",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        default="a.tar",
+        help="output the compiled module to an archive",
+    )
+    parser.add_argument(
+        "--sanitize-diagnostics",
+        action="store_true",
+        default=True,
+        dest="sanitize_diagnostics",
+        help="enable diagnostic sanitization",
+    )
+    parser.add_argument(
+        "--no-sanitize-diagnostics",
+        action="store_false",
+        dest="sanitize_diagnostics",
+        help="disable diagnostic sanitization",
+    )
+    parser.add_argument(
+        "--target",
+        type=parse_target,
+        action="append",
+        metavar="TARGET[,TARGET]...",
+        help=f"compilation target(s): {', '.join(VALID_TARGETS)}, default llvm",
+    )
+    parser.add_argument("--tuner-file", default="", help="tuner file")
+    parser.add_argument(
+        "--alter-layout",
```

Comment:
It could be just `--desire-layout` from users' perspective.

Reply:
"desired layout" is the name being used in the convert layout pass (https://github.com/apache/incubator-tvm/blob/4347b41a5e64a2a453297b371232d6e101051b3c/src/relay/transforms/convert_layout.cc), so I think it should be fine.

If we want to emphasize that this would be applied to data layout only, `--data-layout` might be more precise and sufficient.

Reply:
It is now called `--desired-layout`